packages/tokens-studio-for-figma/src/plugin/setValuesOnVariable.ts

```diff
@@ -7,13 +7,100 @@ import { convertTokenTypeToVariableType } from '@/utils/convertTokenTypeToVariab
 import { checkCanReferenceVariable } from '@/utils/alias/checkCanReferenceVariable';
 import { TokenTypes } from '@/constants/TokenTypes';
 import { transformValue } from './helpers';
+import { ProgressTracker } from './ProgressTracker';
+import { BackgroundJobs } from '@/constants/BackgroundJobs';
+import { notifyUI, postToUI } from './notifiers';
+import { MessageFromPluginTypes } from '@/types/messages';
 
 export type ReferenceVariableType = {
   variable: Variable;
   modeId: string;
   referenceVariable: string;
 };
 
+// Helper function to process tokens in batches to prevent memory leaks
+async function processBatch(
+  batch: SingleToken<true, { path: string, variableId: string }>[],
+  variablesInFigma: Variable[],
+  collection: VariableCollection,
+  mode: string,
+  baseFontSize: string,
+  shouldRename: boolean,
+  variableKeyMap: Record<string, string>,
+  referenceVariableCandidates: ReferenceVariableType[],
+  renamedVariableKeys: string[],
+) {
+  await Promise.all(batch.map(async (token) => {
+    const variableType = convertTokenTypeToVariableType(token.type, token.value);
+    // If id matches the variableId, or name patches the token path, we can use it to update the variable instead of re-creating.
+    // This has the nasty side-effect that if font weight changes from string to number, it will not update the variable given we cannot change type.
+    // In that case, we should delete the variable and re-create it.
+    const variable = variablesInFigma.find((v) => (v.key === token.variableId && !v.remote) || v.name === token.path) || figma.variables.createVariable(token.path, collection, variableType);
```

Comment:
```suggestion
    const variable = variablesInFigma.find((v) => (v.key === token.variableId && !v.remote) || v.name === token.path) ?? figma.variables.createVariable(token.path, collection, variableType);
```
prefer nullish coalescing for fallbacks, so you don't end up using a fallback for a valid falsy value like 0 or ''